main.tf

```
@@ -1 +1,193 @@
+          "containerPath" : "/etc/passwd",
+          "sourceVolume" : "etc_passwd"
+        },
+        {
+          "readOnly" : true,
+          "containerPath" : "/etc/group",
+          "sourceVolume" : "etc_group"
+        }
+      ],
+      "name" : local.ecs_task_family_name,
+      "privileged" : true
+    }
+  )])
+  ecs_service_name     = length(var.ecs_service_name) > 0 ? var.ecs_service_name : "${var.resource_prefix}-service-${random_id.uniq.hex}"
+  ecs_task_family_name = length(var.ecs_task_family_name) > 0 ? var.ecs_task_family_name : "${var.resource_prefix}-task-${random_id.uniq.hex}"
+  iam_role_arn         = var.use_existing_iam_role ? var.iam_role_arn : aws_iam_role.ecs_execution[0].arn
+  iam_role_name        = (!var.use_existing_iam_role && length(var.iam_role_name) > 0) ? var.iam_role_name : "${var.resource_prefix}-role-${random_id.uniq.hex}"
+  ssm_parameter_arn    = (var.use_ssm_parameter_store && !var.use_existing_ssm_parameter) ? aws_ssm_parameter.lacework_access_token[0].arn : var.ssm_parameter_arn
+}
+
+resource "random_id" "uniq" {
+  byte_length = 4
+}
+
+resource "aws_iam_role" "ecs_execution" {
+  count = var.use_existing_iam_role ? 0 : 1
+
+  name = local.iam_role_name
+  tags = var.iam_role_tags
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2008-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "ecs_task_execution_policy_attachment" {
+  count = var.use_existing_iam_role ? 0 : 1
+
+  role       = aws_iam_role.ecs_execution[0].name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+data "aws_iam_policy_document" "ssm_parameter_store_policy" {
+  count = (!var.use_existing_iam_role && var.use_ssm_parameter_store) ? 1 : 0
+
+  version = "2012-10-17"
+
+  statement {
+    sid       = "GetSsmParameter"
+    actions   = ["ssm:GetParameters"]
+    resources = [local.ssm_parameter_arn]
+  }
+
+  dynamic "statement" {
+    for_each = var.ssm_parameter_encrypted ? [1] : []
+    content {
+      sid       = "DecryptSsmParameter"
+      actions   = ["kms:Decrypt"]
+      resources = [var.ssm_parameter_kms_arn]
+    }
+  }
+}
+
+resource "aws_iam_policy" "ssm_parameter_store_policy" {
+  count = (!var.use_existing_iam_role && var.use_ssm_parameter_store) ? 1 : 0
+
+  name        = local.iam_role_name
+  description = "An IAM policy to allow a Lacework Agent ECS task to pull the agent access token from SSM"
+  policy      = data.aws_iam_policy_document.ssm_parameter_store_policy[0].json
+}
+
+resource "aws_iam_role_policy_attachment" "ssm_parameter_store_iam_role_policy" {
+  count = (!var.use_existing_iam_role && var.use_ssm_parameter_store) ? 1 : 0
+
+  role       = local.iam_role_name
+  policy_arn = aws_iam_policy.ssm_parameter_store_policy[0].arn
```

only if you agree 😅:
```suggestion
  policy_arn = aws_iam_policy.ssm_parameter_store_policy[count.index].arn
```